django_x509/static/django-x509/js/switcher.js

```diff
@@ -0,0 +1,53 @@
+(function($){
+  // function for operation_type switcher
+  var filter_div = function(operation_type){
+    //for each div in the form
+    $(".form-row").each(function(i, obj){
```

Comment:
instead of looping on each field and then using conditionals to understand which field to hide or show, you can simply call multiple css selectors, eg:

```javascript
var import_fields = $('.field-certificate, .field-private_key, .field-ecc');

if (operation_type === 'import') {
    import_fields.hide()
}
```

Reply:
Check again this file:
https://github.com/openwisp/django-netjsonconfig/blob/master/django_netjsonconfig/static/django-netjsonconfig/js/switcher.js

See how simple it is? Yes there are less fields to hide/show There's no loop, because there's no need to.

django-loci is much more complex, you can see here multiple items are selected using only CSS selectors:
https://github.com/openwisp/django-loci/blob/master/django_loci/static/django-loci/js/loci.js#L15-L16
These items are then hidden or shown without looping over them.